Add section for Bash Completion (#231)
sb2nov/mac-setup: BashCompletion/README.md

> # Bash Completion

Reviewer: This looks great!

It would be even better if there was a brief description of the "plugin" here, so when you read the page you immediately know why you would want to install it.

Reviewer: @ahmadnurus I'd still like this to be added. What I'm looking for is just a one, maybe two, sentences to describe what this is and why it's useful.

Author: Sorry for my delay, I have added a brief description this bash completion

## Installation

>     $ brew install bash-completion

Reviewer 2: These $ example lines can be in triple-backtick sections with the `console` keyword.

Author: I think it's not a problem using $ as long as it's only one line

> Bash completion will installed on `/usr/local/etc/bash_completion.d`

Reviewer 2: ```suggestion
Bash completion will be installed in `/usr/local/etc/bash_completion.d`
```

> So that this works, Add the following line to your ~/.bash_profile:

Reviewer 2: ```suggestion
For it to work, add this to your `~/.bash_profile`:
```

```bash

> [ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion

Reviewer: Is the `. /usr/local/...` right? With the dot followed by a whitespace. I've never seen it before and I can't find any information about it. What does it do?

Author: That's right.

this is info from brew command

```bash
$ brew info bash-completion
bash-completion: stable 1.3 (bottled)
Programmable completion for Bash 3.2
https://salsa.debian.org/debian/bash-completion
Conflicts with:
  bash-completion@2 (because Differing version of same formula)
/usr/local/Cellar/bash-completion/1.3_3 (189 files, 608.2KB) *
  Poured from bottle on 2018-12-08 at 00:25:02
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/bash-completion.rb
==> Caveats
Add the following line to your ~/.bash_profile:
  [ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion

Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
```

```

Or simply type:

>     $ echo "[ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion" >> ~/.bash_profile

Reviewer: Same question as above.

Restart your bash session:

    $ source ~/.bash_profile

## Usage

> Once you've done this. you can use bash completion by hit `TAB` twice after a command. example:

Reviewer 2: ```suggestion
Once you've done this, you can use bash completion by pressing the tab key twice after a command. For example:
```

```bash
$ git [tab] [tab]
add            blame          cherry-pick    config         format-patch   gui            merge          push           repack         rm             stage          whatchanged
am             branch         citool         describe       fsck           help           mergetool      range-diff     replace        send-email     stash          worktree
apply          bundle         clean          diff           gc             init           mv             rebase         request-pull   shortlog       status
archive        checkout       clone          difftool       gitk           instaweb       notes          reflog         reset          show           submodule
bisect         cherry         commit         fetch          grep           log            pull           remote         revert         show-branch    tag
```

## More

> You can check list additional completion packages are available by typing:

Reviewer: I think a `check` snook into this sentence. It should read `You can list additional ...` right?

    $ brew search completion

> And you can install them using `brew install` command, example:

Reviewer 2: ```suggestion
And you can install them using `brew install` commands, for example:
```

    $ brew install docker-completion

> *You can also add bash completion file into `/usr/local/etc/bash_completion.d` manually*

Reviewer 2: ```suggestion
*You can also manually add a bash completion file into `/usr/local/etc/bash_completion.d`.*
```
